### ISSUE TITLE
Problem with cancellation on source item with negative stock (Magento 2.4.5)

### ISSUE THREAD

**Issue by sunland-jsa** [AmpersandHQ/magento2-disable-stock-reservation#84]
Hi.

We found a cancellation issue when a source item stock is negative (Magento 2.4.5)

1. Order is placed on product with source item in stock.
2. External WMS updates source item to negative stock.
3. Order is canceled.
4. sourceDeductionService fails with "Not all of your products are available in the requested quantity."

Reason for failure is this check in sourceDeductionService->execute (line 88)
**if (($sourceItem->getQuantity() - $qty) >= 0) {**

Please advice :)

**Comment by gianstraf** [comment]
Hi @sunland-jsa,
this pull request should be fix your problem: https://github.com/AmpersandHQ/magento2-disable-stock-reservation/pull/81

We fix this specific problem into the file https://github.com/gianstraf/magento2-disable-stock-reservation/commit/2984795386efec6f5bb2dc77df590ab795d12b82#diff- ff73c03ad62228f6acf854b37caf31cbd87b071520df65cbc06518165909aec1 around rows 86-100

```
/*
 * Fix a problem when a product has a large negative quantity, and an order with that product is canceled from backend.
*/
$salesEvent = $sourceDeductionRequest->getSalesEvent();
if ($salesEvent->getType() == \Magento\InventorySalesApi\Api\Data\SalesEventInterface::EVENT_ORDER_CANCELED) {
    if (($sourceItem->getQuantity() - $qty) < 0) {
        $sourceItem->setQuantity($sourceItem->getQuantity() - $qty);
        $stockStatus = $this->getSourceStockStatus(
            $stockItemConfiguration,
            $sourceItem
        );
        $sourceItem->setStatus($stockStatus);
        continue;
    }
}
```

**Comment by gianstraf** [comment]
@sunland-jsa now this problem should be fixed with the version [1.2.2](https://github.com/AmpersandHQ/magento2-disable-stock-reservation/releases/tag/1.2.2).
